doc/tutorials/notifications.md

```diff
@@ -0,0 +1,327 @@
+
+	// Initialize notification wrapper
+	elektraNotificationOpen (kdb, binding);
+
+	// Start the event loop
+	uv_run (loop, UV_RUN_DEFAULT);
+
+	// Cleanup
+	binding->cleanup (binding);
+	uv_loop_close (loop);
+	free (loop);
+}
+
+void someFunction (void)
+{
+	// notifications are sent asynchronously on kdbSet
+	// over configured transports using our uv event loop
+	kdbSet (repo, key, parentKey);
+}
+```
+ 
+ How to receive notifications
+ ===
+ 
+ We extend the example
+from the previous section where we already created our I/O-binding
+and initialized the notification-wrapper.
+ 
+ In order to handle change
+notifications a developer can either register a variable or a callback.
+Values of registered variables are automatically updated when the value
+of the assigned Key has changed.
+ This is suitable for programs where
+the Key's value is simply displayed or used repeatedly (e.g. by a timer
+or in a loop).
+ When the program's initialization depends on the value
+of the Key (e.g. the value sets the number of worker threads) updating a
+registered variable will not suffice.
+ For these situations a callback
+should be used.
+ 
+ In the following example we will register a integer
+variable:
+ 
+```C
+KDB * repo;
+
+//  ... initialization of KDB and I/O-binding
+
+Key * someKey = keyNew ("/sw/someprogram/v1/someKey", KEY_END);
+int someKeyValue;
+
+elektraNotificationRegisterInt (repo, &someKeyValue, someKey);
+```
+ 
+ The variable `someKeyValue` will be
+automatically updated if the Key in the program above is changed by
+another program (e.g. by using the `kdb` CLI command).
+ 
+ The following
+snippet shows how a callback can be used if the value of the changed Key
+needs further processing.
+ 
+```C
+void myCallback (Key * changedKey)
+{
+		// do something
+}
+
+int main ()
+{
+		KDB * repo;
+
+		// ... initialization of KDB and I/O-binding
+
+		Key * someKey = keyNew ("/sw/someprogram/v1/someKey", KEY_END);
+
+		elektraNotificationRegisterCallback(repo, &myCallback, someKey);
+
+		// ...
+}
+```
+ 
+ How to create your own I/O-binding
+ ===
+ 
+ Developers can
+create their own bindings if the I/O management library of their
+choice is not supported by an existing I/O-binding.
+ 
+ To create a
+binding the I/O-interface's functions pointers need to be filled.
+This interface is a `struct` that contains pointers to functions
+for managing file descriptors, timers and idle operations.
+ The
+interface is independent from I/O management libraries and notification
+transport plugins.
+ See the developer API documentation for details on the
+[ElektraIoInterface](https://doc.libelektra.org/api/current/html/structElektraIoInterface.html).
+
+ 
+ 
+```C
+#include <kdbio.h>
+
+// It is helpful to create a data structure for your binding to store additional data
+typedef struct MyBindingOperationData
+{
+	ElektraIoInterface * binding;
+	// Add additional fields as required
+	char* bar;
+} MyBindingData;
+
+
+/**
+ * Add idle task to IO-Binding
```

Comment:
What is an idle task here? Why do we need to implement it?

Reply:
This is just an example of a function for adding any [timer|idle|fd]-operation - But it's better to add it to the description :)